src/app/ag/page.tsx

```
@@ -1086,6 +1086,113 @@ export default function AGPage() {
 
         const getRegistrationButton = () => {
+            // First, check if user already has a registration - this takes priority over deadline checks
+            if (registrationStatus) {
+                // User has a registration - show appropriate buttons based on status
+                switch (registrationStatus.status) {
+                    case "pending":
+                        // Check if this is a free modality or payment exempt
+                        const needsPayment = userModality && userModality.price > 0 && !userRegistration?.isPaymentExempt;
+                        
+                        if (needsPayment && !registrationStatus.hasReceipt) {
+                            return (
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)